osf/models/collection.py

```diff
@@ -31,11 +37,43 @@ class Meta:
     def _id(self):
         return self.guid._id
 
+    @cached_property
+    def _es_doc_id(self):
+        return '{}-{}'.format(self._id, self.collection._id)
+
+    @classmethod
+    def load(cls, data, select_for_update=False):
+        if isinstance(data, basestring):
```

Comment:
This should also handle the case where `data` is an `int`, and `get` based on `pk`, as the superclass does